src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -105,11 +105,68 @@ private void CreateSwaggerParametersFromParameters(MethodInfo info, List<Paramet
                 var nonBodyAttrs = paramAttrs.Where(x => x.ParamIn != ParameterIn.Body);
                 foreach (var attr in nonBodyAttrs)
                 {
-                    paramsList.Add(new AnnotatedParameter(paramInfo.Name, paramInfo.ParameterType, attr));
+                    paramsList.AddRange(CreateAnnotatedParameter(paramInfo.Name, paramInfo.ParameterType, attr));
                 }
             }
         }
 
+        private IEnumerable<AnnotatedParameter> CreateAnnotatedParameter(string name, Type paramType, RouteParamAttribute routeParamAttribute)
+        {
+            // Non Query Parameter && Primitive parameter type would default to original behavior, which is to return an instance of AnnotatedParameter
+            if (routeParamAttribute.ParamIn != ParameterIn.Query || Primitive.IsPrimitive(paramType))
+            {
+                return new List<AnnotatedParameter> { new AnnotatedParameter(name, paramType, routeParamAttribute) };
+            }
+
+            List<AnnotatedParameter> result = new List<AnnotatedParameter>();
```

Comment:
I think a better name like `annotatedParameters` would be a lot more meaningful.
I find that if I see a variable named `result`, it's often easy to forget what it's actually the result of/for, especially if it's not used directly on the next line.

I also don't like using the word `result` for lists that aren't finalized yet, because it's not quite the "result" yet.